eventcontent.go

```diff
@@ -19,10 +19,33 @@ import (
 	"database/sql/driver"
 	"encoding/json"
 	"fmt"
+	"regexp"
 	"strconv"
 	"strings"
 )
 
+var userIDRegexp *regexp.Regexp
+var roomIDRegexp *regexp.Regexp
+
+func init() {
+	// https://spec.matrix.org/v1.3/appendices/#server-name
+	IPV4AddressExpr := `\d{1,3}\.\d{1,3}.\d{1,3}.\d{1,3}`
+	IPV6AddressExpr := `[\d\x41-\x46\x61-\x66\:\.]{2,45}`
+	dnsNameExpr := `[\w\-\.]{1,255}`
+	portExpr := `\:\d{1,5}`
+	servernameExpr := fmt.Sprint(`(`, IPV4AddressExpr, `|[`, IPV6AddressExpr, `]|`, dnsNameExpr, `)(`, portExpr, `)?`)
```

Comment:
For clarity can we use `fmt.Sprintf` here with a format string, like `fmt.Sprintf(`%s, %s, %s, %s`, ...)`?

Also is there a reason why `IPV6AddressExpr` is surrounded in more square brackets?

Reply:
Yes, that's part of the grammar specified at https://spec.matrix.org/v1.3/appendices/#server-name.
IP v6 addresses should be wrapped in square brackets when there is a prefix or suffix as part of a URI.

I forgot to scape those square brackets, so they are not interpreted as Regex character classes. But now it's fixed.